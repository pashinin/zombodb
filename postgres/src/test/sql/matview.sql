SET datestyle TO 'iso, mdy';

CREATE MATERIALIZED VIEW matview_test AS SELECT * FROM products;
CREATE INDEX idxmatview_test ON matview_test USING zombodb (zdb('matview_test', ctid), zdb(matview_test)) WITH (url='http://localhost:9200/');
SELECT * FROM zdb_estimate_count('matview_test', 'id:*');
SELECT * FROM products WHERE zdb('matview_test', ctid) ==> 'box' ORDER BY id;
DROP MATERIALIZED VIEW matview_test;