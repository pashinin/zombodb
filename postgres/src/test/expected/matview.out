SET datestyle TO 'iso, mdy';
CREATE MATERIALIZED VIEW matview_test AS SELECT * FROM products;
CREATE INDEX idxmatview_test ON matview_test USING zombodb (zdb('matview_test', ctid), zdb(matview_test)) WITH (url='http://localhost:9200/');
WARNING:  No primary key detected for public.matview_test, continuing anyway
SELECT * FROM zdb_estimate_count('matview_test', 'id:*');
 zdb_estimate_count 
--------------------
                  4
(1 row)

SELECT * FROM products WHERE zdb('matview_test', ctid) ==> 'box' ORDER BY id;
 id | name |               keywords               |         short_summary          |                                  long_description                                   | price | inventory_count | discontinued | availability_date 
----+------+--------------------------------------+--------------------------------+-------------------------------------------------------------------------------------+-------+-----------------+--------------+-------------------
  4 | Box  | {wooden,box,"negative space",square} | Just an empty box made of wood | A wooden container that will eventually rot away.  Put stuff it in (but not a cat). | 17000 |               0 | t            | 2015-07-01
(1 row)

DROP MATERIALIZED VIEW matview_test;
